errors/errors.go

```diff
@@ -15,209 +15,389 @@ package errors
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
 
 	"google.golang.org/genproto/googleapis/rpc/errdetails"
-	"google.golang.org/grpc/codes"
+	grpcCodes "google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/runtime/protoiface"
 
-	"github.com/dapr/kit/grpccodes"
+	"github.com/dapr/kit/logger"
 )
 
-const (
-	resourceInfoDefaultOwner = "dapr-components"
-	errorInfoDefaultDomain   = "dapr.io"
-	errorInfoResonUnknown    = "UNKNOWN_REASON"
-)
+var log = logger.NewLogger("dapr.kit")
+
+// Error implements the Error interface and the interface that complies with "google.golang.org/grpc/status".FromError().
+// It can be used to send errors to HTTP and gRPC servers, indicating the correct status code for each.
+type Error struct {
+	// Added error details. To see available details see:
+	// https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto
+	details []proto.Message
+
+	// Status code for gRPC responses.
+	grpcCode grpcCodes.Code
+
+	// Status code for HTTP responses.
+	httpCode int
 
-var UnknownErrorReason = WithErrorReason(errorInfoResonUnknown, codes.Unknown)
+	// Message is the human-readable error message.
+	message string
 
-// ResourceInfo is meant to be used by Dapr components
-// to indicate the Type and Name.
-type ResourceInfo struct {
-	Type  string
-	Name  string
-	Owner string
+	// Tag is a string identifying the error, used with HTTP responses only.
+	tag string
 }
 
-// Option allows passing additional information
-// to the Error struct.
-// See With* functions for further details.
-type Option func(*Error)
+// ErrorBuilder is used to build the error
+type ErrorBuilder struct {
```

Comment:
Does this need to be exported?

Reply:
I guess not, confirming

Reply:
I unexported it in my recent commit, but CI fails: 
```  
Warning: unexported-return: exported func NewBuilder returns unexported type *errors.errorBuilder, which can be annoying to use (revive)
```

So I will re-export it again